calico_kubernetes/calico_kubernetes.py

```diff
@@ -500,44 +521,33 @@ def _get_pod_ports(self, pod):
         return ports
 
     def _get_pod_config(self):
-        """Get the list of pods from the Kube API server."""
-        pods = self._get_api_path('pods')
-        logger.debug('Got pods %s', pods)
-
-        for pod in pods:
-            logger.debug('Processing pod %s', pod)
-            ns = pod['metadata']['namespace'].replace('/', '_')
-            name = pod['metadata']['name'].replace('/', '_')
-            if ns == self.namespace and name == self.pod_name:
-                this_pod = pod
-                break
-        else:
-            raise KeyError('Pod not found: ' + self.pod_name)
-        logger.debug('Got pod data %s', this_pod)
-        return this_pod
+        """Get the pod resource from the API.
+        API Path depends on the KUBE_API_ROOT, namespace, and pod_name
 
-    def _get_api_path(self, path):
-        """Get a resource from the API specified API path.
-
-        e.g.
-        _get_api_path('pods')
-
-        :param path: The relative path to an API endpoint.
-        :return: A list of JSON API objects
-        :rtype list
+        :return: JSON object containing the pod spec
         """
-        logger.info('Getting API Resource: %s from KUBE_API_ROOT: %s',
-                    path, KUBE_API_ROOT)
+        logger.debug('Getting API Resource: from KUBE_API_ROOT: %s',
+                     KUBE_API_ROOT)
         session = requests.Session()
-        if self.auth_token:
+        if self._api_root_secure() and self.auth_token:
             session.headers.update({'Authorization':
                                     'Bearer ' + self.auth_token})
-        response = session.get(KUBE_API_ROOT + path, verify=False)
-        response_body = response.text
 
-        # The response body contains some metadata, and the pods themselves
-        # under the 'items' key.
-        return json.loads(response_body)['items']
+        # Ensure that the API_ROOT is a path to a directory,
+        # then join with the namespaced pod path.
+        root = KUBE_API_ROOT + ('/' if KUBE_API_ROOT[-1:] != '/' else '')
+        path = 'namespaces/%s/pods/%s' % (self.namespace, self.pod_name)
+        full_path = urljoin(root, path)
+        try:
+            response = session.get(full_path, verify=False)
+            logger.debug('Recieved API Response Object: %s', response)
+        except BaseException, e:
```

Comment:
This isn't quite right.

session.get(...) is not going to throw an error like you think it will.  If the request gets an error response (e.g. 404 not found, some other HTTP error), it is going to return successfully, but the `Response` object is going to have an error code.

We need to check that error code and act accordingly.

e.g (pseudo-code)
```
try:
  resp = session.get(url)
except SomeException:
  logger.exception("Exception querying API")
  sys.exit(1)
else:
    if resp.code != 200:
      log("Received error from API response: %s", resp)
      sys.exit(1)
```